src/test/webapp/app/job/my-positions/my-positions-page.component.spec.ts

```diff
@@ -0,0 +1,292 @@
+  });
+
+  it('should navigate to edit job', () => {
+    component.onEditJob('123');
+    expect(router.navigate).toHaveBeenCalledWith(['/job/edit/123']);
+  });
+
+  it('should log error when editing with empty id', () => {
+    const spy = vi.spyOn(console, 'error').mockImplementation(() => {});
+    component.onEditJob('');
+    expect(spy).toHaveBeenCalled();
+  });
+
+  it('should navigate to view job', () => {
+    component.onViewJob('321');
+    expect(router.navigate).toHaveBeenCalledWith(['/job/detail/321']);
+  });
+
+  it('should log error when viewing with empty id', () => {
+    const spy = vi.spyOn(console, 'error').mockImplementation(() => {});
+    component.onViewJob('');
+    expect(spy).toHaveBeenCalled();
+  });
+
+  it('should handle onSearchEmit with new query', async () => {
+    const loadSpy = vi.spyOn(component as unknown as { loadJobs: () => Promise<void> }, 'loadJobs').mockResolvedValue();
+    component.searchQuery.set('old');
+    component.onSearchEmit('new');
+    expect(component.page()).toBe(0);
+    expect(component.searchQuery()).toBe('new');
+    expect(loadSpy).toHaveBeenCalled();
+  });
+
+  it('should not reload if query is same after trim', async () => {
+    const loadSpy = vi.spyOn(component as unknown as { loadJobs: () => Promise<void> }, 'loadJobs').mockResolvedValue();
+    component.searchQuery.set('same query');
+    component.onSearchEmit('  same   query ');
+    expect(loadSpy).not.toHaveBeenCalled();
+  });
+
+  it('should handle filterEmit for job', async () => {
+    const loadSpy = vi.spyOn(component as unknown as { loadJobs: () => Promise<void> }, 'loadJobs').mockResolvedValue();
+    component.onFilterEmit({ filterId: 'job', selectedValues: ['AI'] });
+    expect(component.selectedJobFilters()).toEqual(['AI']);
+    expect(loadSpy).toHaveBeenCalled();
+  });
+
+  it('should handle filterEmit for status', async () => {
+    const loadSpy = vi.spyOn(component as unknown as { loadJobs: () => Promise<void> }, 'loadJobs').mockResolvedValue();
+    component.onFilterEmit({ filterId: 'status', selectedValues: ['jobState.draft'] });
+    expect(component.selectedStatusFilters()).toContain('DRAFT');
+    expect(loadSpy).toHaveBeenCalled();
+  });
+
+  it('should handle loadOnSortEmit correctly', async () => {
+    const loadSpy = vi.spyOn(component as unknown as { loadJobs: () => Promise<void> }, 'loadJobs').mockResolvedValue();
+    component.loadOnSortEmit({ field: 'title', direction: 'ASC' });
+    expect(component.sortBy()).toBe('title');
+    expect(component.sortDirection()).toBe('ASC');
+    expect(loadSpy).toHaveBeenCalled();
+  });
+
+  it('should handle loadOnTableEmit with pagination', async () => {
+    const loadSpy = vi.spyOn(component as unknown as { loadJobs: () => Promise<void> }, 'loadJobs').mockResolvedValue();
+    component.loadOnTableEmit({ first: 20, rows: 10 });
+    expect(component.page()).toBe(2);
+    expect(component.pageSize()).toBe(10);
```

Comment:
```suggestion
    const first = 20, rows = 10;
    component.loadOnTableEmit({ first, rows });
    expect(component.page()).toBe(Math.floor(first / rows));
    expect(component.pageSize()).toBe(rows);
```
Replace the hardcoded magic numbers